api/v1alpha1/nicclusterpolicy_webhook.go

```diff
@@ -0,0 +1,297 @@
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+
+/*
+We validate here
+*/
+func (r *NicClusterPolicy) validateNicClusterPolicy() error {
+	var allErrs field.ErrorList
+	// Validate IBKubernetes
+	ibKubernetes := r.Spec.IBKubernetes
+	if ibKubernetes != nil {
+		allErrs = append(allErrs, ibKubernetes.validate(field.NewPath("spec").Child("ibKubernetes"))...)
+	}
+
+	// Validate OFEDDriverSpec
+	ofedDriver := r.Spec.OFEDDriver
+	if ofedDriver != nil {
+		allErrs = append(allErrs, ofedDriver.validateVersion(field.NewPath("spec").Child("ofedDriver"))...)
+	}
+	// Validate RdmaSharedDevicePlugin
+	rdmaSharedDevicePlugin := r.Spec.RdmaSharedDevicePlugin
+	if rdmaSharedDevicePlugin != nil {
+		allErrs = append(allErrs, r.Spec.RdmaSharedDevicePlugin.validateRdmaSharedDevicePlugin(
+			field.NewPath("spec").Child("rdmaSharedDevicePlugin"))...)
+	}
+	// Validate SriovDevicePlugin
+	SriovNetworkDevicePlugin := r.Spec.SriovDevicePlugin
+	if SriovNetworkDevicePlugin != nil {
+		allErrs = append(allErrs, r.Spec.SriovDevicePlugin.validateSriovNetworkDevicePlugin(
+			field.NewPath("spec").Child("sriovNetworkDevicePlugin"))...)
+	}
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(
+		schema.GroupKind{Group: "mellanox.com", Kind: "NicClusterPolicy"},
+		r.Name, allErrs)
+}
+func (dp *DevicePluginSpec) validateSriovNetworkDevicePlugin(fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	var sriovNetworkDevicePluginConfigJSON map[string]interface{}
+	sriovNetworkDevicePluginConfig := *dp.Config
+
+	// Validate if the SRIOV Network Device Plugin Config is a valid json
+	if err := json.Unmarshal([]byte(sriovNetworkDevicePluginConfig), &sriovNetworkDevicePluginConfigJSON); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"), dp.Config,
+			"Invalid json of SriovNetworkDevicePluginConfig"))
+		return allErrs
+	}
+
+	// Load the JSON Schema
+	sriovNetworkDevicePluginSchemaLoader := gojsonschema.NewReferenceLoader(
+		"file://./schema/sriov_network_device_plugin_schema.json")
+	acceleratorJSONSchemaLoader := gojsonschema.NewReferenceLoader(
+		"file://./schema/accelerator_selector_schema.json")
+	netDeviceJSONSchemaLoader := gojsonschema.NewReferenceLoader("file://./schema/net_device_schema.json")
+	auxNetDeviceJSONSchemaLoader := gojsonschema.NewReferenceLoader("file://./schema/aux_net_device_schema.json")
+
+	// Load the Sriov Network Device Plugin JSON Loader
+	sriovNetworkDevicePluginConfigJSONLoader := gojsonschema.NewStringLoader(sriovNetworkDevicePluginConfig)
+
+	// Perform schema validation
+	result, err := gojsonschema.Validate(sriovNetworkDevicePluginSchemaLoader,
+		sriovNetworkDevicePluginConfigJSONLoader)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"), dp.Config,
+			"Invalid json configuration of SriovNetworkDevicePluginConfig"+err.Error()))
+	} else if result.Valid() {
```

Comment:
You can change the order of conditions here to flat out the if statement. Use `else if !result.Valid()` and iterate over errors here. Move the part that is currently inside the `else if result.Valid()` clause outside of the if condition.

Reply:
Done